contracts/managers/TwoAssetStrategyManagerWithConfirmation.sol

```diff
@@ -0,0 +1,227 @@
+    uint256 public signalConfirmationMaxTime;
+    uint256 public lastCrossoverConfirmationTimestamp;
+    address public contractDeployer;
+
+    /*
+     * TwoAssetStrategyManagerWithConfirmation constructor.
+     *
+     * @param  _coreInstance                    The address of the Core contract
+     * @param  _priceTriggerInstance            The address of the PriceTrigger to be used in the strategy         
+     * @param  _allocationPricerInstance        The address of the AllocationPricer to be used in the strategy        
+     * @param  _auctionLibraryInstance          The address of auction price curve to use in rebalance
+     * @param  _baseAssetAllocation             Starting allocation of the Rebalancing Set in baseAsset amount
+     * @param  _auctionTimeToPivot              The amount of time until pivot reached in rebalance
+     * @param  _auctionSpeed                    Time, in seconds, where 1% of prices are explored during auction
+     * @param  _signalConfirmationMinTime       The minimum time, in seconds, confirm signal an be called after the
+     *                                          last initial crossover confirmation
+     * @param  _signalConfirmationMaxTime       The maximum time, in seconds, confirm signal an be called after the
+     *                                          last initial crossover confirmation
+     */
+    constructor(
+        ICore _coreInstance,
+        IPriceTrigger _priceTriggerInstance,
+        IAllocationPricer _allocationPricerInstance,
+        IAuctionPriceCurve _auctionLibraryInstance,
+        uint256 _baseAssetAllocation,
+        uint256 _auctionTimeToPivot,
+        uint256 _auctionSpeed,
+        uint256 _signalConfirmationMinTime,
+        uint256 _signalConfirmationMaxTime
+    )
+        public
+    {
+        coreInstance = _coreInstance;
+        priceTriggerInstance = _priceTriggerInstance;
+        allocationPricerInstance = _allocationPricerInstance;
+        auctionLibraryInstance = _auctionLibraryInstance;
+        baseAssetAllocation = _baseAssetAllocation;
+        auctionTimeToPivot = _auctionTimeToPivot;
+        auctionSpeed = _auctionSpeed;
+        signalConfirmationMinTime = _signalConfirmationMinTime;
+        signalConfirmationMaxTime = _signalConfirmationMaxTime;
+        contractDeployer = msg.sender;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * This function sets the Rebalancing Set Token address that the manager is associated with.
+     * Since, the rebalancing set token must first specify the address of the manager before deployment,
+     * we cannot know what the rebalancing set token is in advance. This function is only meant to be called 
+     * once during initialization by the contract deployer.
+     *
+     * @param  _rebalancingSetTokenInstance       The address of the rebalancing Set token
+     */
+    function initialize(
+        IRebalancingSetToken _rebalancingSetTokenInstance
```

Comment:
Should probably check that the current component of the Set works with the `BinaryAllocationPricer`

Reply:
I'm gonna check to make sure that the currentSet is one of the collateral sets on the `AllocationPricer`